### PR TITLE
Fix: Memory leak

### PR DESCRIPTION
**Pull request by CalMWolfs**
## What

WIP: Thunderblade says stuff needs to change a bit

Fix memory leaks caused by some entities not being removed properly.
There may still be other negative performance impacts due to mob detection but this fixes the memory leak of the player not being removed from MobData.players leading to the whole previous world still being stored.

## Changelog Fixes
+ Fixed memory leak. - CalMWolfs

## Changelog Technical Details
+ Clear mob detection data more often. - CalMWolfs
    * Clear on config value update and world switch.

### CODE REVIEW

**Review by hannibal002 — CHANGES_REQUESTED** [review]
it breaks the contract that every Spawn Event also has a Despawn event

File: src/main/java/at/hannibal2/skyhanni/data/mob/MobDetection.kt [hunk 1]
```diff
@@ -313,8 +302,7 @@ class MobDetection {
             // is S0EPacketSpawnObject -> addEntityUpdate(packet.entityID)
             is S37PacketStatistics -> // one of the first packets that is sent when switching servers inside the BungeeCord Network (please some prove this, I just found it out via Testing)
             {
-                shouldClear.set(true)
-                allEntitiesViaPacketId.clear()
+                clearMobs()
```
Comment (Thunderblade73) [inline src/main/java/at/hannibal2/skyhanni/data/mob/MobDetection.kt:305]: ehm, you break thread safety here. Will lead to an ConcurrentModifactionExpection.

File: src/main/java/at/hannibal2/skyhanni/data/mob/MobDetection.kt [hunk 2]
```diff
@@ -354,4 +344,21 @@ class MobDetection {
...
+            clearMobs()
+        }
+    }
+
+    private fun clearMobs() {
+        mobDetectionReset()
+        MobData.previousEntityLiving.clear()
+        MobData.currentEntityLiving.clear()
+        MobData.retries.clear()
+        MobData.entityToMob.clear()
```
Comment (Thunderblade73) [inline src/main/java/at/hannibal2/skyhanni/data/mob/MobDetection.kt:359]: this breaks the contract.
Better way is to check if it is empty else show a error with the data.

**Review by Thunderblade73 — CHANGES_REQUESTED** [review]
I like that the config is handled correctly. But the general clearing literally thows out an error, also breaks the contract that every spawn has to have a despawn. And I'm unsure what breaks because of the WorldSwitchEvent, the other event is only there when you leave hypixel, which btw is not handled anymore because of that change.

**Comment by CalMWolfs** [comment]
Closing as memory leak was already fixed in #1473

**Pull request closed without merging**